crates/modelardb_server/src/storage/data_transfer.rs

```diff
@@ -190,14 +198,29 @@ impl DataTransfer {
         );
 
         // Merge the files and transfer them to the remote object store.
-        StorageEngine::merge_compressed_apache_parquet_files(
+        CompressedDataManager::merge_compressed_apache_parquet_files(
             &self.local_data_folder,
             &object_metas,
             &self.remote_data_folder,
             &format!("{COMPRESSED_DATA_FOLDER}/{table_name}/{column_index}"),
         )
         .await?;
 
+        // Delete the metadata for the transferred files from the metadata database.
+        let compressed_files_to_delete: Vec<Uuid> =
+            CompressedFile::object_metas_to_compressed_file_names(object_metas.clone())
```

Comment:
Maybe remove `clone()` by passing the data by reference.

Reply:
Fixed.